climpred/metrics.py

```diff
@@ -193,17 +281,18 @@ def _ppp(ds, control, comparison, running=None, reference_period=None):
         ppp_skill (xarray object): skill of PPP.
 
     """
-    supervector_dim = 'svd'
-    forecast, reference = comparison(ds, supervector_dim)
-    mse_skill = _mse(forecast, reference, dim=supervector_dim)
-    var = _get_variance(
-        control, time_length=running, reference_period=reference_period)
+    mse_skill = _mse(forecast, reference, dim=dim)
+    # dirty implementation
```

Comment:
There seems to be a lot of these "dirty implementations" maybe you could dump this into a function?

```
def _avg_dim(ds, dim):
    if dim in ds:
        ds = ds.mean(dim)
    return ds
```

Reply:
`var = _avg_dim(reference.std('lead'))`

Reply:
definately